### ISSUE TITLE
Operator (config-daemon) not instantiating VFs on master (4.8.0 OK)

### ISSUE THREAD

**Issue by Omar007** [k8snetworkplumbingwg/sriov-network-operator#134]
While not explicitly supported, using 4.8.0 I've had no problem using the SRIOV operator for a Chelsio T580 NIC. Checking with the latest images however, the VFs were no longer instantiated and the resources are no longer available.  

Checking the logs of the different SRIOV pods show that an attempt is made to retrieve the name. After that, nothing, no VFs.  
Tracing it back, this seemed to originate in the config-daemon pod. The name resolution fails because said NIC does not have a `./nic` directory in its sys directory. This in itself shouldn't technically be a problem that would block SR-IOV from functioning but unlike before, it now seems to completely stop processing the device when this happens.  

Doing a quick search for the error in the repo leads me to commit 092394bf773a2152c7f4a6c7aa8409efa89003ff which may have something to do with it.  

Overriding this specific image back to 4.8.0 and deploying with that version of the config-daemon resolves the issue and the NIC works again (even with all other images still on master/latest).

**Comment by Omar007** [comment]
Turns out it's most likely something else and the cause is somewhere higher up the chain; when I manually instantiate the VFs the operator does pick it up as at that point it does get a `./net` directory. Looking into this further..

**Comment by pliurh** [comment]
> Small sidenote, I noticed this operator has 2 repositories (here and Openshift). I'm assuming this is the new upstream location and the report is in the right repo?

Yes, this one is the new upstream, the openshift one is downstream of this repo.

**Comment by pliurh** [comment]
Did you bind vfio driver to the PF? We need a way to find out which devices are network devices if not using the `./net` directory? Any suggestion?

**Comment by Omar007** [comment]
I did not bind the vfio-pci driver to the PF. The PF is actively used by the host system (cxgb4 driver).  
It looked like the code already filtered out NICs by device class before hitting the device names thing but apparently that was too wide and included too many devices so the presence check for the net directory was added? Turns out that check is too strict I guess.  
I do not really have a suggestion from my end other than maybe somehow using the information in the device/node policies to start out with instead of trying to build a list of everything of a certain type by scanning and processing/filtering all PCIe devices first based on other information? I have no idea how realistic that is though and if the information defined in those policies would be available at this stage in the operator.
